nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -156,6 +157,87 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
             pin_memory=config.get('pin_memory', False),
         )
 
+    def _setup_pseudo_label_dataloader(
+        self,
+        manifest_filepaths: Union[List[List[str]], str],
+        tarred_audio_filepaths: Union[List[List[str]], str] = None,
+        batch_size: int = 64,
+    ):
+        """
+        Setup function for a data loader for unlabeled dataset
+
+        Args:
+            manifest_filepaths: Manifests containing information of unlabeled dataset. For tarred dataset manifests should be sharded
+            tarred_audio_filepaths: Tarr audio files which should correspond to manifest files.
+            batch_size: batch size to use during inference.
+
+        Returns:
+            A DataLoader for the given audio file(s).
+        """
+        if self.cfg.train_ds.get('is_tarred', False):
+            dl_config = {
+                'manifest_filepath': manifest_filepaths,
+                'tarred_audio_filepaths': tarred_audio_filepaths,
+                'sample_rate': self.preprocessor._sample_rate,
+                'labels': OmegaConf.to_container(self.decoder.vocabulary),
+                'is_tarred': True,
+                'use_lhotse': True,
```

Comment:
Yes, for tarred datasets, pseudo-label generation is done using Lhotse dataloaders. This approach avoids hangs caused by WebDataset when updating only a part of pseudo-labeled manifests.

Reply:
Are docs updated with SimPL?

Reply:
What docs do you refer to?

Reply:
Lhotse is still experimental so we want it to be explicitly raised. This isn't an issue since Lhotse needs explicit changes to config anyhow. 

Is the hang with WebDataset still current or has it been resolved?

Reply:
Lhotse is not experimental anymore :) IMO it's OK to default to it.